test/malli/generator_test.cljc

```diff
@@ -40,6 +40,17 @@
                     :qualified-symbol]]
       (is (every? (partial m/validate schema) (mg/sample schema {:size 1000})))))
 
+  (testing "double properties"
+    (is (pos? (count (filter (fn [x]
+                               (or (#?(:clj Double/isNaN
+                                       :cljs js/isNaN)
+                                    x)
+                                   (#{##Inf ##-Inf} x)))
+                             (mg/sample [:double
+                                         {:infinite? true
```

Comment:
instead of reserving new plain keywords for generation, I suggest that these keys are namespaced with `gen`, e.g. `:gen/infinite?` and we could have an utility fn `-gen-properties` in `malli.generator` that filters out all keywords that start with `gen` and remove the ns, e.g. `[:double {:gen/infinite? true, :gen/kikka "kukka"}]` would emit `{:infinite? true, :kikka "kukka"}`.

Reply:
`:gen/infinite?` and `:gen/NaN?` defaulting to their corresponding unqualified keywords or ultimately to `false` (current behavior) is a good idea as it matches the behavior in `-min-max`.

However, is there a rationale for having a general purpose utility that extends this to arbitrary properties? Isn't it best to let generator preparing functions pick what they need? (No firm opinion on the matter, just thinking about performance and avoiding unnecessary looping, albeit small)

Reply:
both ways are ok as:

- haven't seen any other need for generic `:gen` properties -> custom thing here is ok
- so the looping would happen only when a generator is created, so the actual generator would still be fast

btw, there are also `-double-gen`, which is used with `:>` etc.

Reply:
All right, I will favor the first option to keep it simple (no over-engineering unless needed :) )

I didn't think too much about other gen utils that rely on `-double-gen` because those schemas do not target doubles specifically (eg. `:>` validates all numbers). The fact they generate some looks more like an implementation detail.